pkg/controller/sparkapplication/sparkapp_metrics_test.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2018 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sparkapplication
+
+import (
+	"net/http"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSparkAppMetrics(t *testing.T) {
+	http.DefaultServeMux = new(http.ServeMux)
+	// Invalid label.
```

Comment:
What does this comment mean?

Reply:
So prometheus doesn't support "-" in metric or label names. Since we were specifying "-" in our workQueue names, I added a helper method to convert "-" to "_" and not error out (https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/pull/227/files#diff-48f67873a59be740236d811d032ce6f6R33) .. This test just validates that we are correctly handling that case as well.